### PR TITLE
[BACK-1737] Delete blobs, data, data sources, restricted tokens and provider sessions when users are deleted

### PR DESCRIPTION
**Pull request by toddkazakov**

### CODE REVIEW

File: auth/events/events.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	logger.Infof("Deleting provider sessions for user")
+	if err := u.client.DeleteAllProviderSessions(u.ctx, payload.UserID); err != nil {
+		errs = append(errs, err)
+		logger.WithError(err).Error("unable to delete provider sessions for user")
+	}
+
+	if len(errs) != 0 {
+		return errors.New("Unable to auth data for user")
```
Comment (derrickburns) [inline auth/events/events.go:44]: Is this error message correct?

File: auth/service/service/service.go [hunk 2]
```diff
@@ -39,6 +45,18 @@ func New() *Service {
 	}
 }
 
+func (s *Service) Run() error {
+	errs := make(chan error, 0)
```
Comment (derrickburns) [inline auth/service/service/service.go:49]: Does this leak channels, or is it called only once?

File: blob/service/service.go [hunk 3]
```diff
@@ -33,6 +39,18 @@ func New() *Service {
...
+func (s *Service) Run() error {
+	errs := make(chan error, 0)
+	go func() {
+		errs <- s.userEventsHandler.Run(context.Background())
+	}()
+	go func() {
+		errs <- s.Service.Run()
+	}()
+
+	return <-errs
```
Comment (derrickburns) [inline blob/service/service.go:51]: Same issue about leaks.

File: auth/service/service/service.go (hunk 2 above)
Comment (toddkazakov) [reply auth/service/service/service.go:49]: Called once on startup

File: blob/service/service.go (hunk 3 above)
Comment (toddkazakov) [reply blob/service/service.go:51]: Called once on startup

File: auth/events/events.go (hunk 1 above)
Comment (toddkazakov) [reply auth/events/events.go]: Fixed

**Review by derrickburns — APPROVED** [review]
Lgtm

**Pull request merged**